packages/fixed-decimal/src/FixedDecimal.ts

```diff
@@ -0,0 +1,471 @@
+   */
+  public toString() {
+    const str = this.value.toString().padStart(this.decimalPlaces + 1, '0')
+    return this.decimalPlaces > 0
+      ? `${str.substring(0, str.length - this.decimalPlaces)}.${str.substring(
+          str.length - this.decimalPlaces
+        )}`
+      : str
+  }
+
+  /**
+   * Analogous to Number().toLocaleString(), with some important differences in
+   * the options available and the defaults. Be sure to check the defaults.
+   * @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat Mozilla NumberFormat documentation}
+   *
+   * @param locale The string specifying the locale (default is 'en-US').
+   * @param options The options for formatting. The available options and defaults are different than NumberFormat.
+   */
+  public toLocaleString(locale?: string, options?: FixedDecimalFormatOptions) {
+    // Apply defaults
+    options = {
+      ...defaultFormatOptions(this),
+      ...options
+    }
+    // Apply rounding method
+    let str = ''
+    switch (options.roundingMode) {
+      case 'ceil':
+        str = this.ceil(options.maximumFractionDigits).toString()
+        break
+      case 'floor':
+        str = this.floor(options.maximumFractionDigits).toString()
+        break
+      case 'trunc':
+        str = this.trunc(options.maximumFractionDigits).toString()
+        break
+      case 'halfExpand':
+        str = this.round(options.maximumFractionDigits).toString()
+        break
+    }
+
+    let [whole, decimal] = str.split('.')
+
+    // Strip trailing zeros
+    decimal = (decimal ?? '').replace(/0+$/, '')
+
+    if (options.minimumFractionDigits !== undefined) {
+      if (
+        options.trailingZeroDisplay !== 'stripIfInteger' ||
+        BigInt(decimal) !== BigInt(0)
+      ) {
+        decimal = decimal.padEnd(options.minimumFractionDigits, '0')
+      }
+    }
+
+    // Localize with a decimal to extract the separator
+    const wholeInt = BigInt(whole)
+    const wholeWithDecimal = wholeInt.toLocaleString(locale, {
+      ...options,
+      minimumFractionDigits: 1,
+      maximumFractionDigits: 1
+    })
+    // Get the separator character
+    const decimalSeparator = wholeWithDecimal.substring(
+      wholeWithDecimal.length - 2,
+      wholeWithDecimal.length - 1
+    )
+    // Remove the decimal
+    whole = wholeWithDecimal.substring(0, wholeWithDecimal.length - 2)
+    return decimal.length > 0 ? `${whole}${decimalSeparator}${decimal}` : whole
+  }
+
+  /**
+   * Formats the decimal as an easy-to-read shorthand summary.
+   * Used primarily for balances in tiles and headers.
+   *
+   * - Always truncates, never rounds up. (eg. `1.9999 => "1.99"`)
+   * - Don't show decimal places if they'd appear as 0. (eg. `1.00234 => "1"`)
+   * - Shows two decimal places if they'd be non-zero. (eg. `1.234 => "1.23"`)
+   * - Count by 1,000s if over 10k (eg. `25413 => "25k"`)
```

Comment:
If I'm not mistaken, our other utility functions for this support the millions suffix as well (`'1.23M'`)

Reply:
They actually don't!